File: src/main/java/de/notjansel/datapacksync/threading/DatapackThread.java
package de.notjansel.datapacksync.threading;

import de.notjansel.datapacksync.Datapacksync;
import org.bukkit.command.CommandSender;

import java.io.IOException;

public class DatapackThread extends Thread {
    private final String[] args;
    private final CommandSender sender;
    public DatapackThread(CommandSender sender, String[] args) {
        this.args = args;
        this.sender = sender;
    }

    @Override
    public void run() {
        if (!sender.hasPermission("datapacksync.use")) {
            sender.sendMessage("You do not have the Permission to use this.");
            return;
        }
        if (args.length == 0) {
            sender.sendMessage("No Filename and URL given, aborting...");
            sender.sendMessage("Syntax: /sync <filename> <url>");
            return;
        }
        if (args[0].equals("debug")) {
            sender.sendMessage(Datapacksync.serverpath);
            return;
        }
        if (args.length == 1) {
            sender.sendMessage("No Url given, aborting...");
            sender.sendMessage("Syntax: /sync <filename> <url>");
            return;
        }
        try {
            Datapacksync.downloadFile(args[1], Datapacksync.serverpath + "/downloads/" + args[0]);
        } catch (IOException e) {
            throw new RuntimeException();
        }
        sender.sendMessage("Download complete. To copy the Datapack, type /copy <Filename>");
    }
}

File: src/main/kotlin/de/notjansel/datapacksync/threading/DatapackThread.kt
package de.notjansel.datapacksync.threading

import de.notjansel.datapacksync.Datapacksync
import org.bukkit.command.CommandSender
import java.io.IOException

class DatapackThread(private val sender: CommandSender, private val args: Array<String>) : Thread() {
    override fun run() {
        if (!sender.hasPermission("datapacksync.use")) {
            sender.sendMessage("You do not have the Permission to use this.")
            return
        }
        if (args.isEmpty()) {
            sender.sendMessage("No Filename and URL given, aborting...")
            sender.sendMessage("Syntax: /sync <filename> <url>")
            return
        }
        if (args[1] == "debug") {
            sender.sendMessage(Datapacksync.serverpath!!)
            return
        }
        if (args.size == 2) {
            sender.sendMessage("No Url given, aborting...")
            sender.sendMessage("Syntax: /datasync download <filename> <url>")
            return
        }
        try {
            Datapacksync.downloadFile(args[2], Datapacksync.serverpath + "/downloads/" + args[1])
        } catch (e: IOException) {
            throw RuntimeException()
        }
        sender.sendMessage("Download complete. To Copy the Datapack, type /datasync copy <Filename>")
    }
}